packages/markers/src/browser/problem/problem-decorations-provider.ts

```diff
@@ -35,13 +36,15 @@ export class ProblemDecorationsProvider implements DecorationsProvider {
 
     @postConstruct()
     protected init(): void {
-        this.problemManager.onDidChangeMarkers(() => {
-            const newUris = Array.from(this.problemManager.getUris(), stringified => new URI(stringified));
-            this.onDidChangeEmitter.fire(newUris.concat(this.currentUris));
-            this.currentUris = newUris;
-        });
+        this.problemManager.onDidChangeMarkers(() => this.fireDidDecorationsChanged());
     }
 
+    private fireDidDecorationsChanged = debounce(() => {
+        const newUris = Array.from(this.problemManager.getUris(), stringified => new URI(stringified));
+        this.onDidChangeEmitter.fire(newUris.concat(this.currentUris));
+        this.currentUris = newUris;
+    }, 10);
```

Comment:
Suggestion: A delay of 10ms is fairly short, and might not yield the expected improvements on a poor network connection. We should increase this delay to 50ms, which is in line with most other delays that we have in the frontend.